compositor_pipeline/src/pipeline.rs

```diff
@@ -58,14 +62,22 @@ pub struct PipelineOutput<Receiver: PipelineOutputReceiver> {
 }
 
 impl<Receiver: PipelineOutputReceiver> PipelineOutput<Receiver> {
-    fn new(opts: OutputOptions<Receiver>) -> Result<Self, String> {
+    fn new(opts: OutputOptions<Receiver>) -> Result<Self, OutputInitError> {
         let mut encoder = Encoder::new(opts.encoder_settings, opts.resolution)?;
         let (tx, rx) = crossbeam_channel::unbounded();
         let (id_tx, id_rx) = crossbeam_channel::bounded(1);
 
         std::thread::spawn(move || {
-            let mut receiver = Receiver::new(opts.receiver_options, encoder.codec());
-            id_tx.send(receiver.identifier()).unwrap();
+            let mut receiver = match Receiver::new(opts.receiver_options, encoder.codec()) {
+                Ok(r) => r,
+                Err(e) => {
+                    id_tx.send(Err(e)).unwrap();
+                    return;
+                }
+            };
```

Comment:
`unwrap_or_else`

Reply:
this is unwrap on sending to channel, it can only happen if other thread panics and mutex is posioned

Reply:
Nevermind, haven't thought about return